app/src/main/java/com/strv/movies/database/AuthDataStore.kt

```diff
@@ -0,0 +1,68 @@
+package com.strv.movies.database
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.MutablePreferences
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStoreFile
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val SESSION_TOKEN = "accessToken"
+private const val AUTH_DATASTORE = "auth_datastore"
+
+
+@Singleton
+class AuthDataStore @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+    companion object{
+        val SESSION_TOKEN_KEY = stringPreferencesKey(SESSION_TOKEN)
+    }
```

Comment:
I would make it **private** and keep it on file level, instead companion object - to be consist with MoviesDataStore